### PR TITLE
:seedling: Improve fd selection logic

### PR DESCRIPTION
**Pull request by rikatz**
**What this PR does / why we need it**:
As the cluster.spec.failureDomainSelector is a real K8s label selector, it can be used when listing the deployment zones directly.

This way, instead of having a full list to iterate and check if the label match on every interaction, we can be already sure the returned list contains only the deployment zones we care about 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by rikatz** [comment]
ok I have no clue what's this all about :D

**Comment by rikatz** [comment]
or maybe I do and I need to do it better on the options vs nil, will check again in a bit

**Comment by rikatz** [comment]
/retest
will try once more and if this doesn't work will verify tomorrow what am I doing wrong :)

File: hack/tools/Makefile [hunk 1]
```diff
@@ -99,7 +99,7 @@ golangci-lint: $(GOLANGCI_LINT)
 $(GOLANGCI_LINT): $(REPO_ROOT)/.github/workflows/golangci-lint.yaml
 	$(REPO_ROOT)/hack/ensure-golangci-lint.sh \
 		-b $(BIN_DIR) \
-		$(shell cat .github/workflows/golangci-lint.yaml | grep version | sed 's/.*version: //')
```
Comment (sbueringer) [inline hack/tools/Makefile:102]: Let's drop this from this PR. This was changed on main to a totally different implementation
(also not sure if we can assume that yq is always available)

**Comment by sbueringer** [comment]
@rikatz lgtm in general. Please drop the change to the Makefile then we should be ready for merge

**Comment by rikatz** [comment]
@sbueringer sorry, missed the notification of this one.

Requested change is done :)

**Comment by rikatz** [comment]
It is a valid error, I need to check it here

**Comment by rikatz** [comment]
/retest

This is a flake :)

**Comment by rikatz** [comment]
@randomvariable @sbueringer aaaand it's done, sorry for the delay on this one

**Pull request merged**